tests/TestCase/Validation/ValidationTest.php

```diff
@@ -1102,6 +1103,7 @@ public function testDateMmddyyyy()
         $this->assertFalse(Validation::date('11.31.2006', ['mdy']));
         $this->assertFalse(Validation::date('11/31/2006', ['mdy']));
         $this->assertFalse(Validation::date('11 31 2006', ['mdy']));
+        $this->assertFalse(Validation::date("12-27-2006\n", ['mdy']));
```

Comment:
My concern is that these are no longer dates. They are dates + a new line. If provided to a database or external system this _could_ fail as there is additional whitespace.